tests/xharness/Log.cs

```diff
@@ -3,134 +3,19 @@
 using System.Diagnostics.Contracts;
 using System.IO;
 using System.Text;
+using xharness.Logging;
 
 namespace xharness
 {
-	public abstract class Log : TextWriter
-	{
-		public Logs Logs { get; private set; }
-		public string Description;
-		public bool Timestamp = true;
-
-		public static string XML_LOG = "XML log";
-		public static string NUNIT_RESULT = "NUnit results";
-		public static string SYSTEM_LOG = "System log";
-		public static string COMPANION_SYSTEM_LOG = "System log (companion)";
-		public static string BUILD_LOG = "Build log";
-		public static string TEST_LOG = "Test log";
-		public static string EXTENSION_TEST_LOG = "Extension test log";
-		public static string EXECUTION_LOG = "Execution log";
-
-		protected Log (Logs logs)
-		{
-			Logs = logs;
-		}
-
-		protected Log (Logs logs, string description)
-		{
-			Logs = logs;
-			Description = description;
-		}
-
-		public abstract string FullPath { get; }
-
-		internal protected virtual void WriteImpl (string value)
-		{
-			Write (Encoding.UTF8.GetBytes (value));
-		}
-
-		public virtual void Write (byte [] buffer)
-		{
-			Write (buffer, 0, buffer.Length);
-		}
-
-		public virtual void Write (byte[] buffer, int offset, int count)
-		{
-			throw new NotSupportedException ();
-		}
-
-		public virtual StreamReader GetReader ()
-		{
-			throw new NotSupportedException ();
-		}
-
-		public override void Write (char value)
-		{
-			WriteImpl (value.ToString ());
-		}
-
-		public override void Write (string value)
-		{
-			if (Timestamp)
-				value = DateTime.Now.ToString ("HH:mm:ss.fffffff") + " " + value;
-			WriteImpl (value);
-		}
-
-		public override void WriteLine (string value)
-		{
-			Write (value + "\n");
-		}
-
-		public override void WriteLine (string format, params object [] args)
-		{
-			Write (string.Format (format, args) + "\n");
-		}
-
-		public override string ToString ()
-		{
-			return Description;
-		}
-
-		public override void Flush ()
-		{
-		}
-
-		public override Encoding Encoding {
-			get {
-				return System.Text.Encoding.UTF8;
-			}
-		}
-
-		public static Log CreateAggregatedLog (params Log [] logs)
-		{
-			return new AggregatedLog (logs);
-		}
-
-		// Log that will duplicate log output to multiple other logs.
-		class AggregatedLog : Log
-		{
-			Log [] logs;
-
-			public AggregatedLog (params Log [] logs)
-				: base (null)
-			{
-				this.logs = logs;
-			}
-
-			public override string FullPath => throw new NotImplementedException ();
-
-			internal protected override void WriteImpl (string value)
-			{
-				foreach (var log in logs)
-					log.WriteImpl (value);
-			}
 
-			public override void Write (byte [] buffer, int offset, int count)
-			{
-				foreach (var log in logs)
-					log.Write (buffer, offset, count);
-			}
-		}
-	}
-
-	public class LogFile : Log
+	public class LogFile : Log, ILogFile
```

Comment:
I think this file should be called LogFile and probably be moved to Logging directory too

Reply:
Yes, that is coming in a second PR. I try to minimise the number of lines to make it simpler for reviews.